doc/man7/EVP_PKEY-DSA.pod

```diff
@@ -31,21 +82,150 @@ The private key value.
 
 =item "p" (B<OSSL_PKEY_PARAM_FFC_P>) <unsigned integer>
 
-A DSA or Diffie-Hellman "p" value.
+A DSA or Diffie-Hellman prime "p" value.
 
 =item "q" (B<OSSL_PKEY_PARAM_FFC_Q>) <unsigned integer>
 
-A DSA or Diffie-Hellman "q" value.
+A DSA or Diffie-Hellman prime "q" value.
 
 =item "g" (B<OSSL_PKEY_PARAM_FFC_G>) <unsigned integer>
 
-A DSA or Diffie-Hellman "g" value.
+A DSA or Diffie-Hellman generator "g" value.
+
+=item "seed" (B<OSSL_PKEY_PARAM_FFC_SEED>) <octet string>
+
+An optional domain parameter I<seed> value used during generation and validation
+of I<p>, I<q> and canonical I<g>.
+For validation this needs to set the I<seed> that was produced during generation.
+
+=item "gindex" (B<OSSL_PKEY_PARAM_FFC_GINDEX>) <integer>
+
+Sets the index to use for canonical generation and verification of the generator
+I<g>.
+Set this to a positive value from 0..FF to use this mode. This I<gindex> can
+then be reused during key validation to verify the value of I<g>. If this value
+is not set or -1 then unverifiable generation of generator I<g> will be used.
+
+=item "pcounter" (B<OSSL_PKEY_PARAM_FFC_PCOUNTER>) <integer>
+
+An optional domain parameter I<counter> value that is output during generation
+of I<p>. This value must be saved if domain parameter validation is required.
+
+=item "hindex" (B<OSSL_PKEY_PARAM_FFC_H>) <integer>
+
+For unverifiable generation of generator I<g> this value is output during 
+generation of I<g>. It value is the first integer larger than one that satisfies
+g = h^j mod p (where g != 1 and "j" is the cofactor).
+
+=item "j" (B<OSSL_PKEY_PARAM_FFC_COFACTOR>) <unsigned integer>
+
+An optional informational cofactor parameter that should equal (p - 1) / q.
+
+=back
+
+
+=head2 DSA / DH key generation (FFC) parameters
+
+The following Key Generation types are available for the built-in FFC algorithms:
+
+=over 4
+
+=item "type" (B<OSSL_PKEY_PARAM_FFC_TYPE>) <utf8_string>
+
+Sets the type of parameter generation. For DH Valid values are:
+
+=over 4
+
+=item "fips186_4"
+
+The current standard. This is the default value.
+
+=item "default"
+
+This is an alias to use the latest implemented standard.
+It is currently set to "fips186_4".
+
+=item "group"
+
+If specified then a named safe prime named will be chosen using the "pbits" type.
```

Comment:
Suggestion: This specifies that a named safe prime name will be chosen...

If ignored, instead change the second _named_ to _name_.

Reply:
Fixed